### PR TITLE
Use Java 11

### LINKED ISSUE #547: Migrate to Java 11
Lets migrate to Java 11. 

Any objections?

### PR DESCRIPTION
**Pull request by mickaelistria**
Fixes #547

### CODE REVIEW

**Comment by cdietrich** [comment]
@eclipse-lsp4j-bot run tests

**Comment by cdietrich** [comment]
https://github.com/eclipse/lsp4j/blob/4bad511eb9471bef9aafc9ad8e7d9c55b6747a25/releng/build.Jenkinsfile#L10

**Comment by cdietrich** [comment]
@eclipse-lsp4j-bot run tests

**Comment by jonahgraham** [comment]
> @eclipse-lsp4j-bot run tests

That won't work as the PR job doesn't use Jenkinsfile. Landing #706 first may help here.

**Comment by cdietrich** [comment]
@jonahgraham as our pr build do this "run on top thing"
is there any way to test this besides pushing it as real branch?

**Comment by cdietrich** [comment]
pushed to https://ci.eclipse.org/lsp4j/job/lsp4j-multi-build/job/mickaelistria-java-11/

**Comment by jonahgraham** [comment]
> pushed to https://ci.eclipse.org/lsp4j/job/lsp4j-multi-build/job/mickaelistria-java-11/

That is a good solution. Alternatively change the JDK in the PR job now.

**Comment by cdietrich** [comment]
https://ci.eclipse.org/lsp4j/job/lsp4j-multi-build/job/mickaelistria-java-11/ is green
@jonahgraham are you ok with the pr?

**Comment by szarnekow** [comment]
> No description provided.

@mickaelistria can you please explain the problem you're going to solve here from an lsp4j maintainers point of view? If I understand it correctly, this change will only make the next release incompatible to Java8 without any other user observable modifications?

**Comment by mickaelistria** [comment]
On #705 , I was encouraged to switch to Java 11 by @jonahgraham who referred to #547 . Supporting less versions is naturally reducing the target/scope of the project and thus reducing the amount of combination that are supposedly supported, thus decreasing the support cost; moving to newer versions enables to use newer feature languages which are most often more powerful or less error-prone, allow to keep the code more modern and more appealing to new contributors.
The real question is not "why" but "why not"? Is anyone in LSP4J committer really gaining anything with Java 8 support? If there is no value generated by this constraint, why not getting rid of it?

**Comment by jonahgraham** [comment]
@eclipse-lsp4j-bot run tests

**Comment by szarnekow** [comment]
@mickaelistria  A link in the commit message would have rendered my question obsolete, I guess.

**Comment by jonahgraham** [comment]
> @mickaelistria A link in the commit message would have rendered my question obsolete, I guess.

I added the missing "Fixes..."

File: releng/build.Jenkinsfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ pipeline {
   }
   tools {
     maven 'apache-maven-latest'
-    jdk 'oracle-jdk8-latest'
+    jdk 'openjdk-jdk11-latest'
```
Comment (jonahgraham) [inline releng/build.Jenkinsfile:10]: openjdk latest is not a good idea, it is not kept up to date. Use adopt instead.

**Review by jonahgraham — CHANGES_REQUESTED** [review]
Needs changelog entry.

I am making the remaining changes now, so no need for more updates

File: releng/build.Jenkinsfile (hunk 1 above)
Comment (nixel2007) [reply releng/build.Jenkinsfile:10]: You mean temurin? :)

Comment (cdietrich) [reply releng/build.Jenkinsfile:10]: yes `temurin-jdk11-latest`

**Comment by jonahgraham** [comment]
@eclipse-lsp4j-bot run tests

**Comment by ice1000** [comment]
Coool!!

**Pull request merged**